### ISSUE TITLE
golint ignores variable clobbering

### ISSUE THREAD

**Issue by chillum** [golang/lint#193]
Hi,

This is potentially unsafe code that is overwriting the variables:

```go
import (
    "encoding/json"
    "fmt"
    "runtime"
)

type ver struct {
	App  string
	Os   string
}

var version ver

func main() {
    version.App = "1.0"
    version.Os = runtime.GOOS

    json, _ := json.Marshal(&version) // [1]
    fmt.Println(string(json))
}
```

[1] – At this point global `json` variable is overwritten and I cannot use `json.*` functions anymore.

It would be very convenient if `golint` detected and reported such situations.

**Comment by dsymonds** [comment]
This isn't a job for golint. It's not a matter of style, but correctness. It's not unsafe either; you'll simply get a compile error if you were trying to access the `json` package, at which point you'll fix it by renaming the var.
